R/endpoints.R

```diff
@@ -0,0 +1,46 @@
+api <- function() {
+  pr <- plumber$new()
+  pr$handle("POST", "/validate", handle_validate)
+  pr
+}
+
+handle_validate <- function(req, pjnz, shape, population, survey, programme,
```

Comment:
The naming scheme through here is going to need some work - it might be clearer if you implemented *two* mini endpoints (even if one is just really trivial like the package version).  I think that we need something like:

```
hintr_response <- function(success, value, error) {
}
```

that all your responses will go through (`prepare_success_response` and `prepare_fail_response` *sound* general but are in fact quite specific).  The `handle_validate` function will need a somewhat better name too - perhaps `pjnz_validate`?  But it might also be worth prefixing these with something, so we might have `endpoint_pjnz_validate` and `endpoint_hint_version` for example - then we'd easily see that these are functions that will definitely return things that are acceptable to the serialiser